crates/asm/common/src/subprotocol.rs

```diff
@@ -122,20 +115,21 @@ pub trait Subprotocol: 'static {
     ///
     /// This is the core transaction processing method where subprotocols implement their
     /// specific business logic. The method receives auxiliary inputs (requested
-    /// during `pre_process_txs`) and can generate messages to other subprotocols and emit logs.
+    /// during `pre_process_txs`) via an [`AuxResolver`] and can generate messages to other
+    /// subprotocols and emit logs.
     ///
     /// # Arguments
     /// * `state` - Mutable reference to the subprotocol's state
     /// * `txs` - Slice of L1 transactions relevant to this subprotocol
     /// * `anchor_pre` - The previous anchor state for validation context
-    /// * `aux_inputs` - Auxiliary data previously requested and validated
+    /// * `aux_resolver` - Interface for retrieving auxiliary data previously requested
     /// * `relayer` - Interface for sending messages to other subprotocols and emitting logs
     /// * `params` - Subprotocol's current params
     fn process_txs(
         state: &mut Self::State,
         txs: &[TxInputRef<'_>],
         anchor_pre: &AnchorState,
-        aux_input: &Self::AuxInput,
+        aux_resolver: &dyn AuxResolver,
```

Comment:
We shouldn't be using `dyn` unless we absolutely have to.

If we're giving up on making aux inputs generic and saying "they have to be one of these types" (defined in the enum), then we might as well just pass a vec of them here instead.